lib/engine_wrapper.py

```diff
@@ -922,16 +924,17 @@ def get_opening_explorer_move(li: LICHESS_TYPE, board: chess.Board, game: model.
                               ) -> tuple[Optional[str], chess.engine.InfoDict]:
     """Get a move from lichess's opening explorer."""
     wb = "w" if board.turn == chess.WHITE else "b"
-    time_left = msec(game.state[f"{wb}time"])
+    time_left = msec(game.state["wtime"]) if board.turn == chess.WHITE else msec(game.state["btime"])
```

Comment:
Why was this change made (and similar changes throughout like line 984)?

Reply:
Because `game.state` only supports `wtime` and `btime` and not all `{text}time` options. We know that `wb` has a value of `"w"` or `"b"`, so `f"{wb}time"` will be either `wtime` or `btime`, but mypy doesn't know this. I could attempt to fix this using [literals](https://mypy.readthedocs.io/en/stable/literal_types.html#parameterizing-literals), like this.
```
# OPTION A
wb: Literal["w", "b"] = "w" if if board.turn == chess.WHITE else "b"
time_left = msec(game.state[f"{wb}time"])

# OPTION B
wbtime: Literal["wtime", "btime"] = "wtime" if if board.turn == chess.WHITE else "btime"
time_left = msec(game.state[wbtime])
```
I don't know if these options work, but I can try later.

Reply:
If the type is `Literal` then the assigned value has to be a literal. This works:
```python
    is_white = board.turn == chess.WHITE
    wbtime: Literal["wtime", "btime"] = "wtime" if is_white else "btime"
    time_left = msec(game.state[wbtime])
```
Since we need `wtime` and `btime` everywhere, I think a function would work better and keep the code simple with many fewer `Literal` declarations:
```python
def wbtime(board: chess.Board) -> Literal["wtime", "btime"]:
    return "wtime" if board.turn == chess.WHITE else "btime"
```
Similar functions for `winc`/`binc` and others would be useful, as well.

Reply:
I added the function for `wbinc` in `engine_wrapper` even though it's not used there, just so it is next to `wbtime`.